### PR TITLE
Enhanced custom business logic DX

### PR DESCRIPTION
**Pull request by renebrandel**
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/gen2/build-a-backend/data/custom-business-logic/index.mdx [hunk 1]
```diff
@@ -4,24 +4,17 @@ export const meta = {
...
-
-**Coming soon:** The custom business logic experience in Amplify (Gen 2) is still under development. Expect the API surface to change dramatically during the Gen-2 preview period. If you have any feedback on the desired developer experience, please [file a GitHub issue](https://github.com/aws-amplify/amplify-backend).
-
-</Callout>
-
-The default GraphQL API provides a solid foundation for querying, mutating, and fetching data. Even so, you may need additional customization to tailor it to your app's specific requirements around data handling, response formatting, and more.
+The `a.model()` data model provides a solid foundation for querying, mutating, and fetching data. However, you may need additional customizations to meet specific requirements around custom API requests, response formatting, and or fetching from external data sources.
 
-In the following sections, we walk through the three steps to create a custom mutation or query:
+In the following sections, we walk through the three steps to create a custom query or query:
```
Comment (palpatim) [inline src/pages/gen2/build-a-backend/data/custom-business-logic/index.mdx:17]: Did you mean to remove "mutation" altogether, or was this an accidental duplication?

File: src/pages/gen2/build-a-backend/data/custom-business-logic/index.mdx [hunk 2]
```diff
@@ -155,10 +154,57 @@ const schema = a.schema({
...
+
+  likePost: a
+    .query()
+    .arguments({ postId: a.id() })
+    .returns(a.ref('Post'))
+    .authorization([a.allow.private()])
+    .handler(a.handler.custom({
+      dataSource: a.ref('Post'),
+      codePath: './increment-like.js'
+    }))
```
[the remark this answers is not in the document]
Comment (renebrandel) [reply src/pages/gen2/build-a-backend/data/custom-business-logic/index.mdx:207]: yes

File: src/pages/gen2/build-a-backend/data/custom-business-logic/index.mdx [hunk 3]
```diff
@@ -171,70 +217,106 @@ export const data = defineData({
...
-  functions: {
-    // Step 3 - Map the function key to the function definition
-    echoHandler: defineFunction({
-      entry: './echo-handler.ts'
-    })
+});
+```
+
+```ts
+// amplify/data/increment-like.js
```
[the remark this answers is not in the document]
Comment (renebrandel) [reply src/pages/gen2/build-a-backend/data/custom-business-logic/index.mdx:224]: no. it'll fail at the CFN level.

**Comment by renebrandel** [comment]
Closing in favor of: https://github.com/aws-amplify/docs/pull/7120

**Pull request closed without merging**